assignments/prototypes.js

```diff
@@ -102,9 +129,57 @@
   console.log(archer.greet()); // Lilith offers a greeting in Elvish.
   console.log(mage.takeDamage()); // Bruce took damage.
   console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
-*/
+
 
   // Stretch task: 
   // * Create Villian and Hero constructor functions that inherit from the Humanoid constructor function.  
   // * Give the Hero and Villians different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
-  // * Create two new objects, one a villian and one a hero and fight it out with methods!
\ No newline at end of file
+  // * Create two new objects, one a villian and one a hero and fight it out with methods!
+
+function Hero(heroAttrs) {
+  Humanoid.call(this, heroAttrs);
+  this.transformations = heroAttrs.transformations;
+  this.abilities = heroAttrs.abilities;
+  this.equipment = heroAttrs.equipment;
+}
+Hero.prototype = Object.create(Humanoid.prototype);
+
+function Villain(villainAttrs) {
+  Hero.call(this, villainAttrs);
+}
+Villain.prototype = Object.create(Hero.prototype);
+
+const aizen = new Villain({
+  createdAt: new Date(),
+  dimensions: {
+    length: 2,
+    width: 2,
+    height: 3,
+  },
+  hp: 45,
+  name: 'Sosuke Aizen',
+  weapons: ['Kyoka Suigetsu', 'Unknown'],
+  language: 'Japanese',
+  equipment: ['Reiatsu Concealing Cloak'],
+  abilities: ['Kido', 'Shunpo', 'Kanzen Saimin', 'Bankai'],
+  transformations: ['Hogyoku Fusion'],
+});
+
+const ichigo = new Hero({
+  createdAt: new Date(),
+  dimensions: {
+    length: 2,
+    width: 2,
+    height: 3,
+  },
+  hp: 30,
+  name: 'Ichigo Kurosaki',
+  weapons: ['Zangetsu', 'Tensa Zangetsu'],
+  language: 'Japanese',
+  equipment: ['Substitute Shinigami Badge', 'Protective Charm', 'Oken Clothing'],
+  abilities: ['Shunpo', 'Getsuga Tensho', 'Bankai', 'Hollowfication'],
```

Comment:
Consider using an object instead to hold the abilities. In this way, you don't have to know the order of the abilities when calling the abilities in the method (e.g. `this.ablities[1]` is not as convenient `this.abilities.shunpo`). Keys are assumed to be camelCased versions of ability name.